beater/beater.go

```diff
@@ -59,15 +60,21 @@ func New(b *beat.Beat, ucfg *common.Config) (beat.Beater, error) {
 		if _, err := regexp.Compile(beaterConfig.Frontend.ExcludeFromGrouping); err != nil {
 			return nil, errors.New(fmt.Sprintf("Invalid regex for `exclude_from_grouping`: %v", err.Error()))
 		}
-		if b.Config != nil && b.Config.Output.Name() == "elasticsearch" {
-			beaterConfig.setElasticsearch(b.Config.Output.Config())
+		if b.Config != nil && beaterConfig.Frontend.SourceMapping.EsConfig == nil {
+			// fall back to elasticsearch output configuration for sourcemap storage if possible
+			if b.Config.Output.Name() == "elasticsearch" {
+				logger.Info("Falling back to elasticsearch output for sourcemap storage")
+				beaterConfig.setSmapElasticsearch(b.Config.Output.Config())
+			} else {
+				logger.Info("Unable to determine sourcemap storage, sourcemaps will not be applied")
```

Comment:
I still think in this case we shouldn't allow users to upload sourcemaps...

Reply:
I agree that the sourcemap upload should fail, I'd even say not failing is a bug in the current implementation (and has nothing to do with this PR).
But I suggest to add `beaterConfig.Frontend.isEnabled() && smapper != nil` at the `killSwitchHandler` check at https://github.com/elastic/apm-server/blob/master/beater/handlers.go#L235

Reply:
@jalvz opened #1128 where we can fight to the death instead.